File: Day60/DSA/Incomplete_Array.java
Question:-  https://www.geeksforgeeks.org/problems/incomplete-array3859/1

Code:-

// User function Template for Java

class Solution {
    int countElements(int[] arr) {
        // code here
        int n = arr.length;
        
        int max = arr[0];
        
        int min = arr[0];
        
        for(int i : arr)
        {
            max = Math.max(i,max);
            min = Math.min(i,min);
        }
        
        int total = max - min + 1;
        
        HashSet<Integer> st = new HashSet<>();
        
        for(int i : arr)
        {
            st.add(i);
        }
        
        return total - st.size();
    }
}

File: Day60/DSA/Incomplete_Array.cpp
Question:-  https://www.geeksforgeeks.org/problems/incomplete-array3859/1

Code:-

// User function template for C++

class Solution {
  public:
    int countElements(vector<int> &arr) {
        // code here
        
        int n = arr.size();
        
        int maxi = arr[0];
        
        int mini = arr[0];
        
        for(int i = 0;i<arr.size();i++)
        {
            maxi = max(maxi,arr[i]);
        }
        
        for(int i = 0;i<arr.size();i++)
        {
            mini = min(mini,arr[i]);
        }
        
        unordered_set<int> s(arr.begin(), arr.end());
        
        int total = maxi - mini + 1;
        
        return total - s.size();
    }
};